src/stream/ws/ws.c

```diff
@@ -97,9 +99,12 @@ static ssize_t stream_ws_process_ping(struct stream_ws* ws, size_t offset)
 
 	int payload_len = MIN(ws->read_index, ws->header.payload_length);
 
-	// Feed back the payload:
-	stream_tcp_send(&ws->base, rcbuf_from_mem(ws->read_buffer + offset,
-			payload_len), NULL, NULL);
+	// Feed back the (unmasked) payload:
+	rcbuf = rcbuf_new(malloc(payload_len), payload_len);
+	assert(rcbuf->payload);
+	ws_copy_payload(&ws->header, rcbuf->payload, ws->read_buffer + offset,
+			payload_len);
+	stream_tcp_send(&ws->base, rcbuf, NULL, NULL);
 
 	stream_ws_advance_read_buffer(ws, payload_len, offset);
 	return 0;
```

Comment:
More issues with the original code here. This should return the number of bytes read.

Reply:
no, this is fine, since there is no data returned from the stream.

Reply:
Separate changes should be in separate commits. Please remember that next time.